packages/client/components/ActivityLibrary/CreateNewActivity/CreateNewActivity.tsx

```diff
@@ -0,0 +1,263 @@
+      {title: 'Create', description: 'takeaway tasks'}
+    ]
+  },
+  {
+    title: 'Estimate Items',
+    type: 'poker' as ActivityType,
+    includedCategories: ['estimation'],
+    image: estimatedEffortTemplate,
+    isEnabled: false,
+    badges: [{category: 'estimation', theme: 'bg-tomato-500 text-white', title: 'Esimation'}],
+    phases: [
+      {title: 'Select', description: 'or create issues to score'},
+      {title: 'Vote', description: 'on 1 or many scoring dimensions'},
+      {title: 'Push', description: 'esimations to your backlog'}
+    ]
+  }
+]
+
+const CategoryTitle = (props: ComponentPropsWithoutRef<'div'>) => {
+  const {children, className, ...rest} = props
+
+  return (
+    <div
+      className={clsx('p-4 text-lg font-semibold leading-5 text-slate-700', className)}
+      {...rest}
+    >
+      {children}
+    </div>
+  )
+}
+
+const query = graphql`
+  query CreateNewActivityQuery {
+    viewer {
+      featureFlags {
+        retrosInDisguise
+      }
+      teams {
+        id
+        tier
+        name
+        ...NewMeetingTeamPicker_selectedTeam
+        ...NewMeetingTeamPicker_teams
+      }
+      availableTemplates(first: 100) @connection(key: "ActivityLibrary_availableTemplates") {
+        edges {
+          node {
+            name
+            teamId
+          }
+        }
+      }
+    }
+  }
+`
+
+interface Props {
+  queryRef: PreloadedQuery<CreateNewActivityQuery>
+}
+
+export const CreateNewActivity = (props: Props) => {
+  const {queryRef} = props
+  const atmosphere = useAtmosphere()
+  const data = usePreloadedQuery<CreateNewActivityQuery>(query, queryRef)
+
+  const {match} = useRouter<{categoryId?: string}>()
+  const {params} = match
+
+  const [selectedActivity, setSelectedActivity] = useState(() => {
+    const defaultActivity = SUPPORTED_CUSTOM_ACTIVITIES[0]!
+    const categoryId = params.categoryId
+    if (!categoryId) return defaultActivity
+
+    const selectedActivity = SUPPORTED_CUSTOM_ACTIVITIES.find((category) =>
+      category.includedCategories.includes(categoryId)
+    )
+    if (!selectedActivity) return defaultActivity
+    return selectedActivity
+  })
+  const [selectedTeam, setSelectedTeam] = useState(sortByTier(data.viewer.teams)[0]!)
+  const {submitting, error, submitMutation, onError, onCompleted} = useMutationProps()
+  const history = useHistory()
+
+  const handleCreateRetroTemplate = () => {
+    if (submitting) {
+      return
+    }
+
+    const teamTemplates = data.viewer.availableTemplates.edges.filter(
+      (template) => template.node.teamId === selectedTeam.id
+    )
+
+    if (teamTemplates.length >= Threshold.MAX_RETRO_TEAM_TEMPLATES) {
+      onError(new Error('You may only have 20 templates per team. Please remove one first.'))
+      return
+    }
+    if (teamTemplates.find((template) => template.node.name === '*New Template')) {
+      onError(new Error('You already have a new template. Try renaming that one first.'))
```

Comment:
not sure how we should handle this error, as there's no way in the UI to `rename template` in this step 🙃

Reply:
+1 could we make sure the server guarantees that the template name is unique when creating it?